src/modm/platform/can/sam-mcan/can.hpp

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2019, 2023, Raphael Lehmann
+ * Copyright (c) 2021, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <optional>
+
+#include <modm/architecture/interface/can.hpp>
+#include <modm/architecture/interface/can_filter.hpp>
+#include <modm/platform/gpio/connector.hpp>
+
+#include "can_bit_timings.hpp"
+
+#include "message_ram.hpp"
+
+namespace {
+using ErrorCallback = void (*)();
+}
+
+namespace modm::platform
+{
+
+extern ErrorCallback mcan0ErrorCallback;
+extern ErrorCallback mcan1ErrorCallback;
+
+/**
+ * @brief		MCAN (CAN with Flexible Data-Rate)
+ *
+ * @author		Raphael Lehmann <raphael@rleh.de>
+ * @author		Christopher Durand <christopher.durand@rwth-aachen.de>
+ * @ingroup		modm_platform_can
+ */
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+class McanDriver : public ::modm::Can
+{
+public:
+	enum class
+	Mode : uint32_t
+	{
+		Normal,
+		Restricted,
+		Monitoring,
+		Sleep,
+		TestExternalLoopback,
+		TestInternalLoopback,
+
+		// for compatibility with bxCAN:
+		ListenOnly = Monitoring, // no acknowledging
+		LoopBack = TestInternalLoopback,
+
+	};
+
+	using ErrorCallback = ::ErrorCallback;
+
+private:
+	static auto*
+	Regs()
+	{
+		if constexpr (id == 0)
+		{
+			return MCAN0;
+		} else if constexpr (id == 1)
+		{
+			#ifdef MCAN1
+			return MCAN1;
+			#else
+			// necessary workaround for devices without MCAN1 peripheral
+			return nullptr;
+			#endif
+		} else
+		{
+			return nullptr;
+		}
+	}
+
+	// helper for static_assert
+	template<class T = void>
+	static inline constexpr bool
```

Comment:
```suggestion
	static constexpr bool
```

`constexpr` implies `inline` here.